pkg/quota/image/registry.go

```diff
@@ -11,19 +11,29 @@
 package image
 
 import (
+	"fmt"
+	"time"
+
 	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/client/cache"
 	"k8s.io/kubernetes/pkg/quota"
 	"k8s.io/kubernetes/pkg/quota/generic"
 
 	osclient "github.com/openshift/origin/pkg/client"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+const (
+	imageTTL time.Duration = time.Minute * 30
```

Comment:
Yes, probably that would be required.  Which ones need reflectors though?
Just one?  Or all of them?

On Wed, Mar 23, 2016 at 4:27 AM, Michal Minar <notifications@github.com>
wrote:

> In pkg/quota/image/registry.go
> <https://github.com/openshift/origin/pull/8195#discussion_r57123091>:
>
> >  	"k8s.io/kubernetes/pkg/quota"
> >  	"k8s.io/kubernetes/pkg/quota/generic"
> >
> >  	osclient "github.com/openshift/origin/pkg/client"
> > +	imageapi "github.com/openshift/origin/pkg/image/api"
> > +)
> > +
> > +const (
> > +	imageTTL time.Duration = time.Minute * 30
>
> @smarterclayton <https://github.com/smarterclayton> Images don't change
> that much if at all. Build often takes several minutes to complete. And
> resource quota sync takes 5 minutes by default. If we want to take
> advantage of previous admission checks or quota sync, we should make it at
> least 6 minutes. I'd round it to 10.
>
> What I'm worried more about is that there are 3 separate image caches:
>
>    - in OriginResourceQuota admission plugin
>    - in ImageLimitRange admission plugin
>    - in resource quota controller
>
> Which triples the memory usage and decreases the change for a hit. Shall
> we perhaps add an ImageCache to a MasterConfig and share it between them?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/openshift/origin/pull/8195/files/d9f70e84dc18bde5a6518ac604b8fbc862e67725..56bf992b580cca4a97099df42a37bc921762e684#r57123091>
>